equinix_metal/equinix_metal/models/vrf_virtual_circuit.py

```diff
@@ -43,34 +45,14 @@ class VrfVirtualCircuit(BaseModel):
     port: Optional[InterconnectionPort] = None
     project: Optional[Project] = None
     speed: Optional[StrictInt] = Field(default=None, description="integer representing bps speed")
-    status: Optional[StrictStr] = Field(default=None, description="The status changes of a VRF virtual circuit are generally the same as Virtual Circuits that aren't in a VRF. However, for VRF Virtual Circuits on Fabric VCs, the status will change to 'waiting_on_peering_details' once the Fabric service token associated with the virtual circuit has been redeemed on Fabric, and Metal has found the associated Fabric connection. At this point, users can update the subnet, MD5 password, customer IP and/or metal IP accordingly. For VRF Virtual Circuits on Dedicated Ports, we require all peering details to be set on creation of a VRF Virtual Circuit. The status will change to `changing_peering_details` whenever an active VRF Virtual Circuit has any of its peering details updated.")
+    status: Optional[VrfVirtualCircuitStatus] = None
     subnet: Optional[StrictStr] = Field(default=None, description="The /30 or /31 subnet of one of the VRF IP Blocks that will be used with the VRF for the Virtual Circuit. This subnet does not have to be an existing VRF IP reservation, as we will create the VRF IP reservation on creation if it does not exist. The Metal IP and Customer IP must be IPs from this subnet. For /30 subnets, the network and broadcast IPs cannot be used as the Metal or Customer IP.")
     tags: Optional[List[StrictStr]] = None
-    type: Optional[StrictStr] = None
+    type: Optional[VrfIpReservationType] = None
     updated_at: Optional[datetime] = None
     vrf: Vrf
     __properties: ClassVar[List[str]] = ["created_at", "customer_ip", "description", "href", "id", "md5", "metal_ip", "name", "nni_vlan", "peer_asn", "port", "project", "speed", "status", "subnet", "tags", "type", "updated_at", "vrf"]
 
-    @field_validator('status')
-    def status_validate_enum(cls, value):
-        """Validates the enum"""
-        if value is None:
-            return value
-
-        if value not in set(['pending', 'waiting_on_peering_details', 'activating', 'changing_peering_details', 'deactivating', 'deleting', 'active', 'expired', 'activation_failed', 'changing_peering_details_failed', 'deactivation_failed', 'delete_failed']):
-            raise ValueError("must be one of enum values ('pending', 'waiting_on_peering_details', 'activating', 'changing_peering_details', 'deactivating', 'deleting', 'active', 'expired', 'activation_failed', 'changing_peering_details_failed', 'deactivation_failed', 'delete_failed')")
-        return value
-
-    @field_validator('type')
```

Comment:
Need to dig more into whether this `@field_validator` gets called by `model_validate` or not.  Working on an example python script to reproduce the validation failure I'm running into with virtual circuits.

Reply:
Welp, turns out the enum validation was working fine and I was getting errors from the virtual circuit module because I hadn't installed the newer metal-python in my virtualenv.